opencv-glib/image.cpp

```diff
@@ -201,6 +201,274 @@ gcv_drawing_options_new(void)
   return GCV_DRAWING_OPTIONS(g_object_new(GCV_TYPE_DRAWING_OPTIONS, NULL));
 }
 
+/*************************************/
+
+typedef struct {
+  gboolean normalize;
+// const Scalar &borderValue=morphologyDefaultBorderValue()
+// const Size &dstsize=Size()
+  gdouble delta;
+  gdouble psi;
+  gdouble scale;
+  gdouble sigma_y;
+  gint border_type;
+  gint iterations;
+  gint ksize;
+//  gint ktype;
+  gint max_level;
+  GCVPoint anchor;
+// TermCriteria termcrit=TermCriteria(TermCriteria::MAX_ITER+TermCriteria::EPS, 5, 1)
+} GCVImageFilteringOptionsPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE(GCVImageFilteringOptions,
+                           gcv_image_filtering_options,
+                           G_TYPE_OBJECT)
+
+#define GCV_IMAGE_FILTERING_OPTIONS_GET_PRIVATE(object) \
+  static_cast<GCVImageFilteringOptionsPrivate *>(      \
+    gcv_image_filtering_options_get_instance_private(   \
+      GCV_IMAGE_FILTERING_OPTIONS(object)))
+
+// TODO
+enum {
+  PROP_NORMALIZE = 1,
+  PROP_BORDER_VALUE,
+  PROP_DSTSIZE,
+  PROP_DELTA,
+  PROP_PSI,
+  PROP_SCALE,
+  PROP_SIGMA_Y,
+  PROP_BORDER_TYPE,
+  PROP_ITERATIONS,
+  PROP_KSIZE,
+  PROP_KTYPE,
+  PROP_MAX_LEVEL,
+  PROP_ANCHOR,
+};
+
+static void
+gcv_image_filtering_options_get_property(GObject *object,
+                                 guint prop_id,
+                                 GValue *value,
+                                 GParamSpec *pspec)
+{
+  auto priv = GCV_IMAGE_FILTERING_OPTIONS_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_NORMALIZE:
+    g_value_set_boolean(value, priv->normalize);
+    break;
+/* TODO
+  case PROP_BORDER_VALUE:
+    g_value_set_XXX(value, priv->border_value);
+    break;
+  case PROP_DSTSIZE:
+    g_value_set_XXX(value, priv->dstsize);
+    break;
+*/
+  case PROP_DELTA:
+    g_value_set_double(value, priv->delta);
+    break;
+  case PROP_PSI:
+    g_value_set_double(value, priv->psi);
+    break;
+  case PROP_SCALE:
+    g_value_set_double(value, priv->scale);
+    break;
+  case PROP_SIGMA_Y:
+    g_value_set_double(value, priv->sigma_y);
+    break;
+  case PROP_BORDER_TYPE:
+    g_value_set_enum(value, priv->border_type);
+    break;
+  case PROP_ITERATIONS:
+    g_value_set_int(value, priv->iterations);
+    break;
+  case PROP_KSIZE:
+    g_value_set_int(value, priv->ksize);
+    break;
+/*
+  case PROP_KTYPE:
+    g_value_set_int(value, priv->ktype);
+    break;
+*/
+  case PROP_MAX_LEVEL:
+    g_value_set_int(value, priv->max_level);
+    break;
+/*
+  case PROP_ANCHOR:
+    g_value_set_XXX(value, priv->anchor);
```

Comment:
In this part, I would like to implement the following.
Could you advise for that?
(Thank you for helping me. I'm not familiar with C++ pointers, by the way.)

```ruby
options = CV::ImageFilteringOptions.new
options.anchor = CV::Point.new(-1,-1)
#...
options.anchor
```